coffee-warehouse/src/components/MenuNavContainer.vue

```diff
@@ -0,0 +1,156 @@
+</template>
+<script>
+import { Menu } from "@progress/kendo-vue-layout";
+import { provideLocalizationService } from "@progress/kendo-vue-intl";
+
+export default {
+  props: {
+    currentTheme: String,
+  },
+  components: {
+    "kendo-menu": Menu,
+  },
+  mounted() {
+    this.$router.push(this.items[0].data);
+    this.setSelectedState(
+      this.selectedMenuItem,
+      this.currentTheme.toLowerCase()
+    );
+  },
+  inject: {
+    kendoLocalizationService: { default: null },
+  },
+  watch: {
+    currentTheme(currentTheme) {
+      this.setSelectedState(this.selectedMenuItem, currentTheme.toLowerCase());
+    },
+  },
+  data() {
+    return {
+      selectedMenuItem: 1,
+      teamClass: "",
+      dashboardClass: "",
+      profileClass: "",
+      infoClass: "",
+    };
+  },
+  computed: {
+    items() {
+      return [
+        {
+          text: this.teamMessage,
+          cssClass: this.teamClass,
+          cssStyle: {
+            paddingBottom: "10px",
+            paddingTop: "10px",
+            paddingLeft: "5px",
+          },
+          icon: "k-icon k-i-grid",
+          data: {
+            path: "/Team",
+          },
+          id: 1,
+        },
+        {
+          text: this.dashboardMessage,
+          cssClass: this.dashboardClass,
+          cssStyle: {
+            paddingBottom: "10px",
+            paddingTop: "10px",
+            paddingLeft: "5px",
+          },
+          icon: "k-icon k-i-chart-line-markers",
+          data: {
+            path: "/Dashboard",
+          },
+          id: 2,
+        },
+        {
+          text: this.profileMessage,
+          cssClass: this.profileClass,
+          cssStyle: {
+            paddingBottom: "10px",
+            paddingTop: "10px",
+            paddingLeft: "5px",
+          },
+          icon: "k-icon k-i-user",
+          data: {
+            path: "/Profile",
+          },
+          id: 3,
+        },
+        {
+          text: this.infoMessage,
+          cssClass: this.infoClass,
+          cssStyle: {
+            borderTop: "1px solid red",
+            paddingBottom: "10px",
+            paddingTop: "10px",
+            paddingLeft: "5px",
+          },
+          icon: "k-icon k-i-info",
+          data: {
+            path: "/Info",
+          },
+          id: 4,
+        },
+      ];
+    },
+    teamMessage() {
+      return provideLocalizationService(this).toLanguageString("team", "Team");
+    },
+    dashboardMessage() {
+      return provideLocalizationService(this).toLanguageString(
+        "dashboard",
+        "Dashboard"
+      );
+    },
+    profileMessage() {
+      return provideLocalizationService(this).toLanguageString(
+        "profile",
+        "Profile"
+      );
+    },
+    infoMessage() {
+      return provideLocalizationService(this).toLanguageString("info", "Info");
+    },
+  },
+  methods: {
+    setSelectedState(id, theme) {
+      if (id === 1) {
+        this.teamClass = theme + " k-state-selected";

```

Comment:
such code is quite often a 'code smell' - one thing that comes to my mind is to check in the computed property what is the selectedMenuItem and add the selected class depending on it there.

Reply:
In commit https://github.com/telerik/kendo-vue/pull/164/commits/1435fc66434b3d047c30c3f71946b0428626a4ab the logic is simplified to call the updateSelectedMenuItem method when two watchers are changed. Using computed properties throws an unusual error' Write operation failed: computed property is read only.' I think it appears because the values of these computed properties are used in the items computed property.